### PR TITLE
[CU-357bfqc] Submit MP

### PR DESCRIPTION
**Pull request by KOngTNP**

### CODE REVIEW

**Comment by sirateek** [comment]
Task linked: [CU-357bfqc Implement "Submit MP Information API"](https://app.clickup.com/t/357bfqc)

File: .env.example [hunk 1]
```diff
@@ -8,4 +8,5 @@ MYSQL_HOSTNAME=
 MYSQL_PORT=
 MYSQL_USERNAME=
 MYSQL_PASSWORD=
-MYSQL_DATABASE=
\ No newline at end of file
+MYSQL_DATABASE=
+API_KEY=
```
Comment (sirateek) [inline .env.example:12]: Duplicated with the one on line 4
```suggestion
```

File: handler/submitmp.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+func (a *submitmpHandler) SubmitMp(g *gin.Context) {
+	g.BindJSON(&mp)
+	err := a.submitmpService.SubmitMp(mp.CitizenID)
+	if err != nil {
+		g.Status(http.StatusInternalServerError)
+		return
+	}
+	g.Status(http.StatusOK)
+	return
```
Comment (sirateek) [inline handler/submitmp.go:31]: Redundant return statement.

File: models/submitmp.go [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package model
+
+type SubmitMp struct {
+	ID        int    `db:"ID"`
+	CitizenID string `db:"CitizenID"`
+}
```
Comment (sirateek) [inline models/submitmp.go:6]: I'll merge the model into `models`.

My mistake from previous pr. 😅

File: repository/submitmp.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+		mysql: mysql,
+	}
+}
+
+func (a *submitMpRepository) SubmitMpToDB(citizenID string) error {
+	rows, err := a.mysql.Query("INSERT INTO Mp (CitizenID) VALUES (?)", citizenID)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
```
Comment (sirateek) [inline repository/submitmp.go:26]: Redundant defer.

Comment (sirateek) [reply repository/submitmp.go:26]:
```suggestion
	rows.Close()
```

File: repository/submitmp.go [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+	return &submitMpRepository{
+		mysql: mysql,
+	}
+}
+
+func (a *submitMpRepository) SubmitMpToDB(citizenID string) error {
+	rows, err := a.mysql.Query("INSERT INTO Mp (CitizenID) VALUES (?)", citizenID)
+	if err != nil {
+		return err
+	}
```
Comment (sirateek) [inline repository/submitmp.go:25]: We should log the error by using `logrus.Error` for tracing back krub.

File: models/submitmp.go (hunk 3 above)
Comment (sirateek) [reply models/submitmp.go:6]: Fixed

File: repository/population.go [hunk 6]
```diff
@@ -3,7 +3,7 @@ package repository
 import (
 	"fmt"
 
-	"github.com/WorkWorkWork-Team/gov-ec-api/model"
+	model "github.com/WorkWorkWork-Team/gov-ec-api/models"
```
Comment (sirateek) [inline repository/population.go:6]: What is this?

File: service/population.go [hunk 7]
```diff
@@ -1,7 +1,7 @@
 package service
 
 import (
-	"github.com/WorkWorkWork-Team/gov-ec-api/model"
+	model "github.com/WorkWorkWork-Team/gov-ec-api/models"
```
Comment (sirateek) [inline service/population.go:4]: What is this?

File: repository/population.go (hunk 6 above)
Comment (sirateek) [reply repository/population.go]: Fixed

File: service/population.go (hunk 7 above)
Comment (sirateek) [reply service/population.go]: Fixed

**Review by sirateek — APPROVED** [review]

File: handler/submitmp.go [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+func (a *submitmpHandler) SubmitMp(g *gin.Context) {
+	g.BindJSON(&mp)
+	err := a.submitmpService.SubmitMp(mp.CitizenID)
+	if err != nil {
+		g.Status(http.StatusInternalServerError)
+		return
+	}
+	g.Status(http.StatusOK)
+}
```
Comment (abounk) [inline handler/submitmp.go:31]: Add logger for this function

File: service/submitmp.go [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+func NewSubmitmpService(submitmpRepository repository.SubmitMpRepository) SubmitmpService {
+	return &submitmpService{
+		submitmpRepository: submitmpRepository,
+	}
+}
+
+func (a *submitmpService) SubmitMp(citizenID string) error {
+	return a.submitmpRepository.SubmitMpToDB(citizenID)
+}
```
Comment (abounk) [inline service/submitmp.go:23]: Add logger for this function

File: repository/submitmp.go [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+func (a *submitMpRepository) SubmitMpToDB(citizenID string) error {
+	rows, err := a.mysql.Query("INSERT INTO Mp (CitizenID) VALUES (?)", citizenID)
+	if err != nil {
+		logrus.Info("Have error when query that is: ", err)
+		return err
+	}
+	rows.Close()
+	return nil
```
Comment (abounk) [inline repository/submitmp.go:29]: Add logger for this case too

**Review by abounk — APPROVED** [review]

**Pull request merged**